internal/server/apimigrator.go

```diff
@@ -105,50 +105,59 @@ func rewriteRequired(c *gin.Context, migrationVersion *semver.Version) bool {
 func rebuildRequest(c *gin.Context, newReqObj interface{}) {
 	query := url.Values{}
 	body := map[string]interface{}{}
-	r := reflect.ValueOf(newReqObj)
-	t := r.Type()
-	for i := 0; i < r.NumField(); i++ {
-		f := r.Field(i)
-		if fieldName, ok := t.Field(i).Tag.Lookup("form"); ok {
-			if f.Type() == reflect.TypeOf(uid.ID(0)) {
-				query.Add(fieldName, uid.ID(f.Int()).String())
-				continue
+
+	var rebuildRequestQuery func(t reflect.Value)
+
+	rebuildRequestQuery = func(v reflect.Value) {
+		t := v.Type()
+		for i := 0; i < v.NumField(); i++ {
+			f := v.Field(i)
+			if v.Field(i).Kind() == reflect.Struct {
+				// follow the rabbit hole for nested structs
+				rebuildRequestQuery(f)
```

Comment:
This recursively unpacks any nested structs (like `api.PaginationRequest`) so that they will also be included in the `query` and `body` variables.